src/intelExt.cpp

```diff
@@ -0,0 +1,184 @@
+             const long* input,
+             long n,
+             long q) 
+{
+  initNTT(/*degree=*/n, /*modulus=*/q)
+      .ComputeInverse(reinterpret_cast<uint64_t*>(output),
+                      reinterpret_cast<const uint64_t*>(input),
+                      /*input_mod_factor=*/2,
+                      /*output_mod_factor=*/1);
+  return;
+}
+
+void EltwiseAddMod(long* result,
+                   const long* operand1,
+                   const long* operand2,
+                   long n,
+                   long modulus)
+{
+  intel::hexl::EltwiseAddMod(reinterpret_cast<uint64_t*>(result),
+                             reinterpret_cast<const uint64_t*>(operand1),
+                             reinterpret_cast<const uint64_t*>(operand2),
+                             n,
+                             modulus);
+}
+
+void EltwiseAddMod(long* result,
+                   const long* operand,
+                   long scalar,
+                   long n,
+                   long modulus)
+{
+  intel::hexl::EltwiseAddMod(reinterpret_cast<uint64_t*>(result),
+                             reinterpret_cast<const uint64_t*>(operand),
+                             scalar,
+                             n,
+                             modulus);
+}
+
+void EltwiseSubMod(long* result,
+                   const long* operand1,
+                   const long* operand2,
+                   long n,
+                   long modulus)
+{
+  intel::hexl::EltwiseSubMod(reinterpret_cast<uint64_t*>(result),
+                             reinterpret_cast<const uint64_t*>(operand1),
+                             reinterpret_cast<const uint64_t*>(operand2),
+                             n,
+                             modulus);
+}
+
+void EltwiseSubMod(long* result,
+                   const long* operand,
+                   long scalar,
+                   long n,
+                   long modulus)
+{
+  intel::hexl::EltwiseSubMod(reinterpret_cast<uint64_t*>(result),
+                             reinterpret_cast<const uint64_t*>(operand),
+                             scalar,
+                             n,
+                             modulus);
+}
+
+void EltwiseMultMod(long* result,
+                    const long* operand1,
+                    const long* operand2,
+                    long n,
+                    long modulus)
+{
+  intel::hexl::EltwiseMultMod(reinterpret_cast<uint64_t*>(result),
+                              reinterpret_cast<const uint64_t*>(operand1),
+                              reinterpret_cast<const uint64_t*>(operand2),
+                              n,
+                              modulus,
+                              /*input_mod_factor=*/1);
+}
+
+void EltwiseMultMod(long* result,
+                    const long* operand,
+                    long scalar,
+                    long n,
+                    long modulus)
+{
+  // HEXL currently doesnt have a scalar multiply.
+  std::vector<uint64_t> scalar_array(n, scalar);
```

Comment:
[hexl::EltwiseFMAMod](https://github.com/intel/hexl/blob/41106872333f61dcabe9461c66c3a744f919feac/hexl/include/hexl/eltwise/eltwise-fma-mod.hpp) should cover this case.